src/huggingface_hub/keras_mixin.py

```diff
@@ -135,13 +145,181 @@ def _create_model_card(
+            plot_model (:obj:`bool`):
+                Setting this to `True` will plot the model and put it in the model card. Requires graphviz and pydot to be installed.
+            include_optimizer (:obj:`bool`, `optional`):
+                Whether or not to include optimizer during serialization.
+            model_save_kwargs(:obj:`dict`, `optional`):
+                model_save_kwargs will be passed to tf.keras.models.save_model() through save_pretrained_keras().
+        """
+        super().__init__()
+        self.save_strategy = save_strategy
+        self.save_steps = save_steps
+        self.plot_model = plot_model
+        self.include_optimizer = include_optimizer
+        self.task_name = task_name
+        self.last_job = None
+        self.model_save_kwargs = model_save_kwargs
+
+        if repo_path_or_name is None and repo_url is None:
+            raise ValueError(
+                "You need to specify a `repo_path_or_name` or a `repo_url`."
+            )
+
+        if not isinstance(token, str):
+            token = HfFolder.get_token()
+
+        if token is None:
+            raise ValueError(
+                "You must login to the Hugging Face hub in your CLI by typing `huggingface-cli login` and "
+                "entering your token. Alternatively, you can pass your own token as the"
+                "`token` argument or use notebook_login() if you're logging in from a notebook."
+            )
+
+        if repo_path_or_name is None:
+            self.repo_path_or_name = repo_url.split("/")[-1]
+        else:
+            self.repo_path_or_name = repo_path_or_name
+
+        if repo_url is None and not os.path.exists(self.repo_path_or_name):
+            repo_name = Path(self.repo_path_or_name).name
+            repo_url = HfApi(endpoint=api_endpoint).create_repo(
+                repo_name,
+                token=token,
+                organization=organization,
+                repo_type=None,
+                exist_ok=True,
+            )
+
+        self.repo = Repository(
+            self.repo_path_or_name,
+            clone_from=repo_url,
+            use_auth_token=token,
+            git_user=git_user,
+            git_email=git_email,
+        )
+        self.repo.git_pull()
+
+    def on_train_batch_end(self, batch, logs=None):
+        if self.save_strategy == "steps" and (batch + 1) % self.save_steps == 0:
+            if self.last_job is not None and not self.last_job.is_done:
+                while not self.last_job.is_done:
+                    time.sleep(1)
+            save_pretrained_keras(
+                self.model,
+                self.repo_path_or_name,
+                include_optimizer=self.include_optimizer,
+                plot_model=self.plot_model,
+                task_name=self.task_name,
+                override_card=True,
+                **self.model_save_kwargs,
+            )
+            self.repo.git_add(auto_lfs_track=True)
```

Comment:
`push_to_hub` already does `git_add` at https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/repository.py#L1324. Same in other places